### PR TITLE
Feature#14 create company branch controller

### PR DESCRIPTION
**Pull request by OwenLobato**
# Description
- [x] Create views of Stores
-  index
- new
- edit
- show
- [x] Create Stores controller

### CODE REVIEW

**Review by V-Juarez — APPROVED** [review]

File: app/controllers/stores_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      end
+    end
+  
+    private
+    def store_params
+      params.require(:store).permit(:name, :address)
+    end
+  
+  end
+
```
Comment (germanruzca) [inline app/controllers/stores_controller.rb:52]: Try to add a new line correctly

**Review by lucasvidela94 — APPROVED** [review]
👍  nice job!

File: app/views/stores/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<div style="display: flex; flex-direction: column; align-items:center;">
+<h1>Store Information</h1>
+<hr>
+<div style="display:grid; grid-direction:column; justify-content: space-around; gap: 10px; padding-left: 10px;" >
+    <div style="display:flex; flex-direction: column;justify-content: space-between; gap: 10px;">
```
Comment (germanruzca) [inline app/views/stores/show.html.erb:5]: Try to add styles to the stylesheets files

File: app/views/stores/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<hr>
+<h1>STORES</h1>
+<div style="display:grid; grid-template-columns: 1fr 1fr;justify-content: space-evenly; gap: 10px">
```
Comment (germanruzca) [inline app/views/stores/index.html.erb:3]: Try to add styles to the stylesheets files

File: app/views/stores/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  <div>
+    <%= form.label :name %>
+    <%= form.text_field :name %>
+  </div>
+  <div>
+    <%= form.label :address %>
+    <%= form.text_field :address %>
+  </div>
+  <%= form.submit 'Edit store' %>
+<% end %>
```
Comment (germanruzca) [inline app/views/stores/edit.html.erb:13]: New line here

**Review by HikingCarrot7 — DISMISSED** [review]
LGTM!

File: app/views/stores/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    </div>
+    <div>
+    <%= button_to  "Detail", store, method: :get %>
+    </div>
+  <% end %>
+</div>
+<div>
+  <a href="<%= new_store_path %>">
+    <button>Create New Store</button>
+  </a>
```
Comment (germanruzca) [inline app/views/stores/index.html.erb:16]: Just as a recommendation (you can dismiss this) try to use the button_to helper, including the GET method:
 ` <%= button_to 'Create New Store', new_store_path, method: :get %>`
```suggestion
  <%= button_to 'Create New Store', new_store_path, method: :get %>
```

Just a recommendation for future implementations

**Review by germanruzca — CHANGES_REQUESTED** [review]

File: app/controllers/stores_controller.rb (hunk 1 above)
Comment (OwenLobato) [reply app/controllers/stores_controller.rb:52]: Done

**Review by AgustinPalmaM — APPROVED** [review]
LGTM

**Review by germanruzca — DISMISSED** [review]
LGTM

File: app/controllers/stores_controller.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    if @store.update(store_params)
+      redirect_to stores_path, notice: 'Store edited ok'
+    else
+      render :edit, :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @store = Store.find(params[:id])
+    if @store.destroy
```
Comment (juperameza) [inline app/controllers/stores_controller.rb:41]: You should do a logical delete

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by V-Juarez — APPROVED** [review]

**Review by YilKanda — COMMENTED** [review]
Try to keep the issue as simple as possible, that is, if you make a rubocop correction, only the files corresponding to your file should appear corrected.

My suggestion is that if you want to correct all the files with one or more Linters, it is preferable that you indicate it in a separate PR.

**Review by V-Juarez — DISMISSED** [review]

File: app/controllers/stores_controller.rb (hunk 6 above)
Comment (ruruisu) [reply app/controllers/stores_controller.rb:41]: done

File: app/helpers/employees_helper.rb [hunk 7]
```diff
@@ -1,2 +1,4 @@
+# frozen_string_literal: true
+
 module EmployeesHelper
 end
```
[the remark this answers is not in the document]
Comment (ruruisu) [reply app/helpers/employees_helper.rb:4]: done, removed the file from the project

File: app/views/employees/show.html.erb [hunk 8]
```diff
@@ -1,36 +1,38 @@
...
+                <%= @employee.admin %>
+            </div>
+        </div>
+        
+        <button>
+            <%= link_to  "Edit Employee", edit_employee_path, class: "btn btn-inverse" %>
+        </button>
+
+        <%= button_to  "Delete Employee", @employee, method: :delete %>
+    </div>
```
[the remark this answers is not in the document]
Comment (ruruisu) [reply app/views/employees/show.html.erb]: done

File: app/views/stores/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+            <label for="">Address:</label>
+         <%= @store.address %>
+        </div>
+    </div>
+    <button>
+        <%= link_to  "Edit Store", edit_store_path, class: "btn btn-inverse" %>
+    </button>
+    <%= button_to  "Delete Store", @store, method: :delete %>
+    </div>
+</div>
```
[the remark this answers is not in the document]
Comment (ruruisu) [reply app/views/stores/show.html.erb]: done

**Review by V-Juarez — APPROVED** [review]

**Review by juperameza — APPROVED** [review]
LGTM

**Review by AgustinPalmaM — APPROVED** [review]
LGTM

**Pull request merged**